### ISSUE TITLE
[BUG] [HIFI4] PeakVol.Quality issues for audio formats

### ISSUE THREAD

**Issue by gbernatxintel** [thesofproject/sof#9213]
**Describe the bug**
There is a problem processing signals of 11025, 22050, 44100, 88200, 176400 Hz by the HIFI4 module. The main problem is that, for example, for the frequency f=44.1kHz frame is 44 or 45. Some samples are not processed.  

**To Reproduce**
1. Run the test: 
pytest -s "tests/avs/fw_01_base_fw_modules/test_09_peak_vol.py::TestPeakVolFull::test_01_09_peakvol_quality[44100Hz_16in16bit_1ch-0]"

**Reproduction Rate**
100%

**Expected behavior**
The entire signal should be processed correctly and no glitches should occur

**Impact**
It is not possible to use HIFI4, for MTL. Using HIFI3 does not use the full capabilities of the platform. The program runs sub optimally and uses several percent more CPU cycles than when using HIFI4

**Environment**
1) Branch name and commit hash of the 2 repositories: sof (firmware/topology) and linux (kernel driver).
    * Kernel: 53ddff639562ef68dc0a6f62b82f7505c75ebdce
    * SOF: 0bda13d0bdaa953f6d98d759f4e805adfb9fb331
2) Name of the platform(s) on which the bug is observed.
    * Platform: MTL
3) Topology:
```
                +--------------------------------------------------+
        +----+  | +------+   +-------+   +----------+   +-------+  |
        |    |--->|Copier|-->|PeakVol|-->|PCM Copier|-->| MixIn |-----+
        |    |  | +------+   +-------+   +----------+   +-------+  |  |
        |    |  +--------------------------------------------------+  |
        |    |                                                        |
        |HOST|            +----------------------------+              |
        |    |            |  +--------+   +--------+   |              |
        |    | <-------------| Copier |<--| MixOut |<-----------------+
        |    |            |  +--------+   +--------+   |
        |    |            +----------------------------+
        +----+
```
    
**Notes**
A PR has been added that fixes the problem for frequencies of 88.2kHz and above. However, it does not solve the problem completely. You can find the link here:
https://github.com/thesofproject/sof/pull/9158

**Screenshots or console output**

![image](https://github.com/thesofproject/sof/assets/163305076/377a0a94-15ee-4910-9246-3956629c4ca0)

cc:
- #9209

**Comment by marc-hb** [comment]
@gbernatxintel can you please confirm that any other HiFi4 hardware other than MTL is or would be affected in the way? In other words, a pure software bug?

**Comment by gbernatxintel** [comment]
> can you please confirm that any other HiFi4 hardware other than MTL is or would be affected in the same way? In other words, a pure software bug?

The problem is replicated on the LNL platform. The way it works is exactly the same as for MTL. The output signal looks the same for LNL when using HIFI4.

The fix [9208](https://github.com/thesofproject/sof/pull/9208) that is here fixes the problem for the LNL platform

**Comment by marc-hb** [comment]
@gbernatxintel can this be closed?

**Comment by lgirdwood** [comment]
@singalsu can this be closed ?

**Comment by singalsu** [comment]
> @singalsu can this be closed ?

No, I'm working with this issue.

**Comment by singalsu** [comment]
Small progress, I'm now able replicate the issue with Linux & SOF topology in LNL platform.

**Comment by wszypelt** [comment]
@singalsu @gbernatxintel  Good news, I checked the above test, and other similar for 9408 at LNL Everything works fine :)

**Comment by singalsu** [comment]
Fixed by #9408 merge
